### PR TITLE
feat: Support configuring Camel K CR metadata and Pipe errorHandler

### PR DESCRIPTION
**Pull request by igarashitm**
Fixes: #181
Also partially addresses #188, need more work for supporting JSON schema `oneOf`

### CODE REVIEW

File: packages/ui/src/pages/PipeErrorHandler/PipeErrorHandlerPage.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  const schemaMap = useSchemasStore((state) => state.schemas);
+  const entitiesContext = useContext(EntitiesContext);
+  const pipeResource = entitiesContext?.camelResource as PipeResource;
+
+  const errorHandlerSchema = useMemo(() => {
+    return schemaMap['PipeErrorHandler'].schema;
+  }, [schemaMap]);
+
+  const isSupported = useMemo(() => {
+    return pipeResource && pipeResource.getType() in [SourceSchemaType.Pipe, SourceSchemaType.KameletBinding];
```
Comment (lordrip) [inline packages/ui/src/pages/PipeErrorHandler/PipeErrorHandlerPage.tsx:19]: [curious-question] Is this working as expected?

The `in` operator looks for keys in objects, but in this case, we're looking for strings in a number-keyed object (Array)

I guess this should be instead:
```suggestion
    return pipeResource && [SourceSchemaType.Pipe, SourceSchemaType.KameletBinding].includes(pipeResource.getType());
```

**Review by lordrip — APPROVED** [review]
Leaving aside a few suggestions and open questions, LGTM.

File: packages/ui/src/pages/PipeErrorHandler/PipeErrorHandlerPage.tsx (hunk 1 above)
Comment (igarashitm) [reply packages/ui/src/pages/PipeErrorHandler/PipeErrorHandlerPage.tsx:19]: ahhh right, it doesn't work for array while works for enum...
```typescript
const array = ['strA', 'strB', 'strC'];

console.log('strB' in array);  // <-- false

enum Types {
    TypeA, TypeB, TypeC
};

console.log('TypeA' in Types); // <--- true
```

File: packages/ui/src/models/visualization/metadata/metadataEntity.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+  constructor(public parent: MetadataParentType) {}
+
+  toJSON() {
+    return { metadata: this.parent.metadata };
+  }
+
+  updateModel(): void {
+    return;
+  }
```
[the remark this answers is not in the document]
Comment (igarashitm) [reply packages/ui/src/models/visualization/metadata/metadataEntity.ts:21]: Yes through the entities, ATM I haven't yet seen this `updateModel()` is in use, is it for future use?

File: packages/ui/src/models/camel/camel-route-resource.ts [hunk 3]
```diff
@@ -48,11 +48,21 @@ export class CamelRouteResource implements CamelResource {
...
 
-  addEntity(entity: BaseCamelEntity): void {
-    this.entities.push(entity);
-  }
-
   toJSON(): unknown {
     return this.entities.map((entity) => entity.toJSON());
   }
+
+  createBeansEntity(): BeansEntity {
```
[the remark this answers is not in the document]
Comment (igarashitm) [reply packages/ui/src/models/camel/camel-route-resource.ts:55]: Yes valid, someone might want to do for code readability, although there's no functional difference. That's the reason I left `createBeansEntity()` create the new one. On the other hand in Kamelet the `beans` is a declared property on the route template (object), and it's always a single entry.

File: packages/ui/src/models/camel/camel-k-resource.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      };
+      return;
+    }
+    this.metadata = this.resource!.metadata && new MetadataEntity(this.resource!);
+  }
+
+  createMetadataEntity() {
+    this.resource!.metadata = {};
+    this.metadata = new MetadataEntity(this.resource!);
+    return this.metadata;
```
[the remark this answers is not in the document]
Comment (igarashitm) [reply packages/ui/src/models/camel/camel-k-resource.ts:40]: The empty check & remove when it's serialized in `toJSON()`, that sounds like an option :+1:  But we'd need to re-evaluate if the entities `toJSON()` is always leveraged (I suspect not, just the root CR object is used from CamelResource class). I'd like to separate that work even if we go that route.

File: packages/ui/src/models/camel/camel-route-resource.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (lordrip) [reply packages/ui/src/models/camel/camel-route-resource.ts:55]: Alright, at this moment, only the first `Bean` definition is being supported by the `MetadataEditor`

File: packages/ui/src/models/visualization/metadata/metadataEntity.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (lordrip) [reply packages/ui/src/models/visualization/metadata/metadataEntity.ts:21]: Oh, then it's likely to be removed later on, if not used.

**Pull request merged**